examples/ucp_client_server.c

```diff
@@ -99,24 +99,24 @@ static struct {
  */
 static void usage(void);
 
-void buffer_free(ucp_dt_iov_t *iov)
+static void buffer_free(ucp_dt_iov_t *iov, size_t iov_size)
 {
     size_t idx;
 
-    for (idx = 0; idx < iov_cnt; idx++) {
+    for (idx = 0; idx < iov_size; idx++) {
```

Comment:
Since `iov` is initialized with all zeroes, maybe instead of passing iov_size we can just free if element is not NULL?
```C
static void buffer_free(ucp_dt_iov_t *iov)
{
    size_t idx;

    for (idx = 0; idx < iov_cnt; ++idx) {
        if (NULL != iov[idx].buffer) {
            mem_type_free(iov[idx].buffer);
            iov[idx].buffer = NULL;
        }
    }
}
```
Or we can make check for NULL in `mem_type_malloc` itself

Reply:
Yes, we can. However, this option looks more complicated, and adds a condition inside the loop body.